tutorials/param_search_vaecf.ipynb

```diff
@@ -0,0 +1,392 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 18,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Load MovieLens 100K ratings\n",
+    "ml_100k = movielens.load_feedback(variant=\"100K\")\n",
+    "\n",
+    "# Define an evaluation method to split feedback into train, validation and test sets\n",
+    "ratio_split = RatioSplit(data=ml_100k, test_size=0.1, val_size=0.1, seed=123, verbose=False)\n",
+    "\n",
+    "# Instantiate Recall@100 and NDCG@100 for evaluation\n",
+    "rec100 = cornac.metrics.Recall(100)\n",
+    "ndcg100 = cornac.metrics.NDCG(100)\n",
+    "\n",
+    "# Instantiate VACF with fixed hyperparameters\n",
+    "vaecf = cornac.models.VAECF(k=20, h=40, learning_rate=0.005, n_epochs=100, seed=123, verbose=False)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Perform searches for the *beta* parameter"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Assume for now we are interested in determining a good value for the hyperparameter $\\beta$, the weight of the KL term in VAECF's objective:\n",
+    "\n",
+    "$$ \\mathcal{L}(\\theta,\\phi) = \\mathbb{E}_{q_{\\phi}(z|r)}[\\log{p_{\\theta}(r|z)}] - \\beta \\cdot KL(q_{\\phi}(z|r)||p(z)), $$\n",
+    "\n",
+    "where $z$ is the latent variable (user representation), and $r$ is the observed user-item feedback."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Wrap vaecf into search methods\n",
+    "All we need to do is to wrap our instantiated model ``vaecf`` into a search method and specify the search space for *beta*, the metric of interest, as well as the evaluation method. Cornac supports two types of search methods, namely random and grid searches, we consider both of them here."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# GridSearch\n",
+    "gs_vaecf = GridSearch(\n",
+    "    model=vaecf,\n",
+    "    space=[\n",
+    "        Discrete(\"beta\", [0.0, 0.2, 0.4, 0.6, 0.8, 1., 1.2, 1.4, 1.6, 1.8, 2.]),\n",
+    "    ],\n",
+    "    metric=rec100,\n",
+    "    eval_method=ratio_split,\n",
+    ")\n",
+    "\n",
+    "# RandomSearch\n",
+    "rs_vaecf = RandomSearch(\n",
+    "    model=vaecf,\n",
+    "    space=[\n",
+    "        Continuous(\"beta\", low=0.0, high=2.),\n",
+    "    ],\n",
+    "    metric=rec100,\n",
+    "    eval_method=ratio_split,\n",
+    "    n_trails=20,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "As evident from above, there are two ways we can specify search spaces, namely `Discrete` and `Continuous`. In the latter case, `n_trails` values will be uniformly sampled in the interval [low,high). More details in the [documentation](https://cornac.readthedocs.io/en/latest/hyperopt.html)."
```

Comment:
"As evident from above, there are two ways we can specify search spaces, namely `Discrete` and `Continuous`. ~~In the latter case, `n_trails` values will be uniformly sampled in the interval [low,high)~~. More details in the [documentation](https://cornac.readthedocs.io/en/latest/hyperopt.html)."